### PR TITLE
many: move to go modules 

### PR DESCRIPTION
**Pull request by anonymouse64**
* Add a go.mod and go.sum which at least passes unit tests and is buildable on
  Ubuntu with go 1.13.
* Introduce evdev sub-dir with fork of golang-evdev as upstream is currently
  broken when building with modules
* Make the osutil/udev package a separate module so it can be imported properly
  via modules and use a replace directive to use it temporarily.

Opening as a draft since we will probably need to iterate on the packaging changes
a bit before this is actually ready for real.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thank you!

**Comment by mvo5** [comment]
Thanks, this looks great. Fwiw, I think it would be slightly clearer VCS wise to split
```
* Introduce evdev sub-dir with fork of golang-evdev as upstream is currently
  broken when building with modules
* Make the osutil/udev package a separate module so it can be imported properly
  via modules and use a replace directive to use it temporarily.
```
into two distinct commits and also separate them from the other parts. But might now be a bit much work, not sure.

**Comment by anonymouse64** [comment]
Yah unfortunately ci isn't even remotely happy yet so I will need some more force pushes here, but when everything is working then sure I can split up the commits like that

**Comment by anonymouse64** [comment]
Alright the github actions unit tests are happy, onto the spread tests ...

File: packaging/ubuntu-16.04/rules [hunk 1]
```diff
@@ -209,7 +207,7 @@ endif
 	$(MAKE) -C data all
 
 	# build squashfuse and rename to snapfuse
-	(cd vendor/github.com/snapcore/squashfuse/src && mkdir -p autom4te.cache && ./autogen.sh --disable-demo && ./configure --disable-demo && make && mv squashfuse_ll snapfuse)
+	(cd _build/src/$(DH_GOPKG)/vendor/github.com/snapcore/squashfuse/src && mkdir -p autom4te.cache && ./autogen.sh --disable-demo && ./configure --disable-demo && make && mv squashfuse_ll ../../../../vendor/snapfuse)
```
Comment (anonymouse64) [inline packaging/ubuntu-16.04/rules:210]: sadly this does not seem to work, either with this change or with what's in master, would appreciate some help on what is going on with how and why we are building squashfuse/snapfuse like this

Comment (mardy) [reply packaging/ubuntu-16.04/rules]: Do you have logs for the build failure at hand?

**Comment by anonymouse64** [comment]
Closing in favor of https://github.com/snapcore/snapd/pull/10634

**Pull request closed without merging**